io.openems.edge.controller.ess.predictivedelaycharge/src/io/openems/edge/controller/ess/predictivedelaycharge/dc/DcPredictiveDelayCharge.java

```diff
@@ -66,14 +70,22 @@ protected void deactivate() {
 	public void run() throws OpenemsNamedException {
 		// Get required variables
 		ManagedSymmetricEss ess = this.componentManager.getComponent(this.config.ess_id());
-		EssDcCharger charger = this.componentManager.getComponent(this.config.charger_id());
 		Integer calculatedPower = super.getCalculatedPower(ess, productionHourlyPredictor, consumptionHourlyPredictor,
 				componentManager);
 
 		// checking if power per second is calculated
 		if (calculatedPower != null) {
-
-			int productionPower = charger.getActualPower().value().orElse(0);
+			Integer productionPower = 0;
+			for (String chargerId : this.config.charger_ids()) {
+				EssDcCharger charger;
+				try {
+					charger = this.componentManager.getComponent(chargerId);
+				} catch (OpenemsNamedException e) {
```

Comment:
If you catch the error here, you will never find out about it. You'd rather throw it, so that the Controller "RunFailed" StateChannel gets set.

Reply:
corrected. Thanks